google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2922,6 +2962,40 @@ boolean isValid() {
     }
   }
 
+  /**
+   * Returns a multiplexed session. We would always return the session which is at the front of the
+   * queue since this will be the most recently created session.
+   */
+  SessionFuture getMultiplexedSessionWithFallback() throws SpannerException {
+    if (options.getUseMultiplexedSession()) {
+      ISpan span = tracer.getCurrentSpan();
+      span.addAnnotation("Acquiring multiplexed session");
```

Comment:
I don't think we should add any annotations or do anything else special with the current span when getting a multiplexed session. In almost all cases, this will be a non-blocking, very quick operation, and adding an annotation will rather add noise (and a few microseconds of latency) than any real information.

Reply:
Wouldn't customers get confused about the missing spans?

Reply:
* These should all be sub-spans of the actual spans (e.g. for a read-only transactions). So they would still see the main spans, but just miss one sub-span.
* The sub-span would always be a few microseconds, except for the very first few queries. So it would then trigger the question 'why do we include a span for this, but not for (insert any random in-mem operation that takes a few microseconds)?'.

Would it maybe make more sense to only add a span in case the multiplexed session has not yet been created?

Reply:
Yes, sounds good. I'll include a span just for the case where a session could not be acquired.